distributed/scheduler.py

```diff
@@ -6064,19 +6064,35 @@ def handle_long_running(
         We stop the task from being stolen in the future, and change task
         duration accounting as if the task has stopped.
         """
+        if worker not in self.workers:
+            logger.debug(
+                "Received long-running signal from unknown worker %s. Ignoring.", worker
+            )
+            return
+
```

Comment:
This is mostly for good measure, I _think_ it should the code should also work without this.